Modules/xSharePoint/DSCResources/MSFT_xSPShellAdmins/MSFT_xSPShellAdmins.psm1

```diff
@@ -0,0 +1,584 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.String]    $Name,
+        [parameter(Mandatory = $false)] [System.String[]] $Members,
+        [parameter(Mandatory = $false)] [System.String[]] $MembersToInclude,
+        [parameter(Mandatory = $false)] [System.String[]] $MembersToExclude,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance[]] $ContentDatabases,
+        [parameter(Mandatory = $false)] [System.Boolean]  $AllContentDatabases,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    if ($Members -and (($MembersToInclude) -or ($MembersToExclude))) {
+        Write-Verbose -Verbose "Cannot use the Members parameter together with the MembersToInclude or MembersToExclude parameters"
+        return $null
+    }
+
+    if ($ContentDatabases) {
+        foreach ($contentDatabase in $ContentDatabases) {
+            if ($contentDatabase.Members -and (($contentDatabase.MembersToInclude) -or ($contentDatabase.MembersToExclude))) {
+                Write-Verbose -Verbose "ContentDatabases: Cannot use the Members parameter together with the MembersToInclude or MembersToExclude parameters"
+                return $null
+            }
+
+            if (!$contentDatabase.Members -and !$contentDatabase.MembersToInclude -and !$contentDatabase.MembersToExclude) {
+                Write-Verbose -Verbose "ContentDatabases: At least one of the following parameters must be specified: Members, MembersToInclude, MembersToExclude"
+                return $null
+            }
+        }
+    } else {
+        if (!$Members -and !$MembersToInclude -and !$MembersToExclude) {
+            Write-Verbose -Verbose "At least one of the following parameters must be specified: Members, MembersToInclude, MembersToExclude"
+            return $null
+        }
+    }
+
+    if ($ContentDatabases -and $AllContentDatabases) {
+        Write-Verbose -Verbose "Cannot use the ContentDatabases parameter together with the AllContentDatabases parameter"
+        return $null
+    }
+
+    Write-Verbose -Message "Getting all Shell Admins"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+
+        try {
+            $spFarm = Get-SPFarm
+        } catch {
+            Write-Verbose -Verbose "No local SharePoint farm was detected. Health Analyzer Rule settings will not be applied"
```

Comment:
This error still talks about the health analyser rules instead of the shell admins